fendermint/actors/chainmetadata/src/actor.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021-2023 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use cid::Cid;
+use fil_actors_runtime::actor_dispatch;
+use fil_actors_runtime::actor_error;
+use fil_actors_runtime::builtin::singletons::SYSTEM_ACTOR_ADDR;
+use fil_actors_runtime::runtime::{ActorCode, Runtime};
+use fil_actors_runtime::ActorDowncast;
+use fil_actors_runtime::ActorError;
+use fil_actors_runtime::Array;
+use fvm_shared::clock::ChainEpoch;
+use fvm_shared::error::ExitCode;
+
+use crate::BlockHash;
+use crate::{ConstructorParams, Method, PushBlockParams, State};
+
+#[cfg(feature = "fil-actor")]
+fil_actors_runtime::wasm_trampoline!(Actor);
+
+pub struct Actor;
+
+impl Actor {
+    fn constructor(rt: &impl Runtime, params: ConstructorParams) -> Result<(), ActorError> {
```

Comment:
Just to make sure I understand: I don't think we ever call this constructor, right? We insert the state prepared in the genesis interpreter and move on, unlike with the EVM actors.

Reply:
Right, we actually only call `PushBlock` method of this actor, everything else (constructing, and lookup up blockhash by epoch in the get_tipset_cid) is done from the outside.